### DISCUSSION TITLE
Custom Events in React

### DISCUSSION THREAD

**Discussion by cameron-smtm** [tinybirdco/web-analytics-starter-kit#71] [Q&A]
Is there an npm package to consume to enable emitting custom events in a React (and next js) project? The README suggests calling those methods dynamically after the script has been loaded, but how would you do this in other frameworks?

**Comment by radchukd** [comment]
👋 I am not sure, I understand you, can you elaborate, please?
You can send custom event to tinybird using this piece of code:

```
Tinybird.trackEvent('add_to_cart', {
  partnumber: 'A1708 (EMC 3164)',
  quantity: 1,
})
```

to check if the script is loaded you can use `const isTbScriptLoaded = typeof Tinybird !== 'undefined'`
